server/test/test.js

```diff
@@ -844,3 +847,163 @@ describe("Clients API tests:", () => {
     });
   });
 });
+
+// Client Login API TESTS
+describe("Client Login API tests:", () => {
+  beforeEach(() => {
+    dao.open();
+    mongoUnit.load(testData.clientsCollection);
+  });
+
+  afterEach(() => {
+    mongoUnit.drop();
+    dao.close();
+  });
+
+  describe("POST /clients/login", () => {
+    it("it should log the user in ", (done) => {
+      chai
+        .request(app)
+        .post("/api/clients/login")
+        .send({ username: "ehsanansari@gmail.com", password: config.password1 })
+        .end((err, res) => {
+          expect(err).to.be.null;
+          expect(res.status).to.be.equal(200);
+          expect(res.body).to.be.an("object");
+          expect(res.body.email).to.be.equal("ehsanansari@gmail.com");
+          expect(res.body._id).to.be.equal("618d4ad3736f2caf2d3b3ca5");
+          expect(res.body.wallet).to.be.equal(55.5);
+          expect(res.body.address).to.be.equal(
+            "fsfsaf dsafsa fsafsa,26 Milano,12342"
+          );
+
+          done();
+        });
+    });
+
+    it("it must fail when an invalid email is passed", (done) => {
+      chai
+        .request(app)
+        .post("/api/clients/login")
+        .send({
+          email: "nomail",
```

Comment:
It should be the **username** and not the **mail**! As you did in the previous test!
Now it is returning 401 because it is missing the username credential